tests/sentry/notifications/test_notifications.py

```diff
@@ -0,0 +1,377 @@
+        Test that an email AND Slack notification are sent with
+        the expected values when an issue is resolved.
+        """
+        url = f"/api/0/issues/{self.group.id}/"
+        with self.tasks():
+            response = self.client.put(url, format="json", data={"status": "resolved"})
+        assert response.status_code == 200, response.content
+
+        msg = mail.outbox[0]
+        # check the txt version
+        assert f"{self.user.username} marked {self.short_id} as resolved" in msg.body
+        # check the html version
+        assert f"{self.short_id}</a> as resolved</p>" in msg.alternatives[0][0]
+
+        attachment = get_attachment()
+
+        assert attachment["title"] == "Resolved Issue"
+        assert attachment["text"] == f"{self.name} marked {self.short_id} as resolved"
+        assert (
+            attachment["footer"]
+            == f"<http://testserver/organizations/{self.organization.slug}/issues/{self.group.id}/?referrer=ResolvedActivitySlack|{self.short_id}> via <http://testserver/settings/account/notifications/?referrer=ResolvedActivitySlack|Notification Settings>"
+        )
+
+    @responses.activate
+    def test_sends_deployment_notification(self):
+        """
+        Test that an email AND Slack notification are sent with
+        the expected values when a release is deployed.
+        """
+
+        release = self.create_release()
+        url = f"/api/0/organizations/{self.organization.slug}/releases/{release.version}/deploys/"
+        with self.tasks():
+            response = self.client.post(
+                url, format="json", data={"environment": self.environment.name}
+            )
+        assert response.status_code == 201, response.content
+
+        msg = mail.outbox[0]
+        # check the txt version
+        assert f"Version {release.version} was deployed to {self.environment.name} on" in msg.body
+        # check the html version
+        assert (
+            f"Version {release.version} was deployed to {self.environment.name}\n    </h2>\n"
+            in msg.alternatives[0][0]
+        )
+
+        attachment = get_attachment()
+
+        assert (
+            attachment["title"] == f"Deployed version {release.version} to {self.environment.name}"
+        )
+        assert (
+            attachment["text"]
+            == f"Version {release.version} was deployed to {self.environment.name}"
+        )
+        assert (
+            attachment["footer"]
+            == "<http://testserver/settings/account/notifications/?referrer=ReleaseActivitySlack|Notification Settings>"
+        )
+
+    @responses.activate
+    def test_sends_regression_notification(self):
+        """
+        Test that an email AND Slack notification are sent with
+        the expected values when an issue regresses.
+        """
+        # resolve and unresolve the issue
+        ts = time() - 300
+        manager = EventManager(make_event(event_id="a" * 32, checksum="a" * 32, timestamp=ts))
```

Comment:
I just wanna point out that I wrote this test first, then later wrote `test_sends_issue_notification` where I use `self.store_event`. I spent some time trying to re-write this test to use that instead, but couldn't get it to work for regressions. I was copying the pattern [here](https://github.com/getsentry/sentry/blob/0b6bc8a532577b4ab048f522a0617d94cc8d0dd1/tests/sentry/tasks/post_process/tests.py#L487) - I created an event, resolved the group, created another one, and was able to verify that the 2 events belonged to the same group and it went from resolved to unresolved, but it never fired a notification. I spent way too long on all of these so I'm going with what actually passes.